safekeeper/src/pull_timeline.rs

```diff
@@ -139,6 +193,54 @@ pub async fn stream_snapshot_guts(
     Ok(())
 }
 
+impl Timeline {
+    /// Simple snapshot for an offloaded timeline: we will only upload a renamed partial segment and
+    /// pass a modified control file into the provided tar stream (nothing with data segments on disk, since
+    /// we are offloaded and there aren't any)
+    async fn snapshot_offloaded<W: AsyncWrite + Unpin + Send>(
+        self: &Arc<Timeline>,
+        ar: &mut tokio_tar::Builder<W>,
+        source: NodeId,
+        destination: NodeId,
+    ) -> Result<()> {
+        let shared_state = self.write_shared_state().await;
+
+        let mut control_store = TimelinePersistentState::clone(shared_state.sk.state());
```

Comment:
@arssher is the lock acquired in `write_shared_state` sufficient to make it safe to go read the control file?  At this point in the code I have checked that the timeline is offloaded (so it shouldn't be mutating things), but there isn't anything guarding it against getting un-offloaded while this code runs.

Reply:
See `switch_to_present`; WAL file might be concurrently downloaded, but cfile update is guarded.